tests/core/service/stream/stream_service_test.py

```diff
@@ -0,0 +1,208 @@
+from unittest.mock import MagicMock, AsyncMock
+
+import pytest
+
+from symphony.bdk.core.auth.auth_session import AuthSession
+from symphony.bdk.core.service.stream.stream_service import StreamService
+from symphony.bdk.gen import ApiClient, Configuration
+from symphony.bdk.gen.agent_api.share_api import ShareApi
+from symphony.bdk.gen.agent_model.share_content import ShareContent
+from symphony.bdk.gen.pod_api.room_membership_api import RoomMembershipApi
+from symphony.bdk.gen.pod_api.streams_api import StreamsApi
+from symphony.bdk.gen.pod_model.stream_filter import StreamFilter
+from symphony.bdk.gen.pod_model.stream_type import StreamType
+from symphony.bdk.gen.pod_model.v2_admin_stream_filter import V2AdminStreamFilter
+from symphony.bdk.gen.pod_model.v2_room_search_criteria import V2RoomSearchCriteria
+from symphony.bdk.gen.pod_model.v2_stream_attributes import V2StreamAttributes
+from symphony.bdk.gen.pod_model.v3_room_attributes import V3RoomAttributes
+from tests.utils.resource_utils import object_from_json_relative_path
+
+
+@pytest.fixture()
+def auth_session():
+    auth_session = AuthSession(None)
+    auth_session.session_token = 'session_token'
+    auth_session.key_manager_token = 'km_token'
+    return auth_session
+
+
+@pytest.fixture()
+def mocked_api_client():
+    api_client = MagicMock(ApiClient)
+    api_client.call_api = AsyncMock()
+    api_client.configuration = Configuration()
+    return api_client
+
+
+@pytest.fixture()
+def stream_service(mocked_api_client, auth_session):
+    return StreamService(StreamsApi(mocked_api_client), RoomMembershipApi(mocked_api_client),
+                         ShareApi(mocked_api_client), auth_session)
+
+
+@pytest.mark.asyncio
+async def test_get_stream(mocked_api_client, stream_service):
```

Comment:
you can even mock the `StreamsApi`, `RoomMembershipApi` and `ShareApi` to assert if the method was called with the correct params. For instance: https://github.com/SymphonyPlatformSolutions/symphony-api-client-python/blob/2.0/tests/core/service/user/user_service_test.py#L77-L82